bundles/org.openhab.binding.weatherunderground/src/main/java/org/openhab/binding/weatherunderground/internal/handler/WeatherUndergroundHandler.java

```diff
@@ -553,7 +551,9 @@ private boolean updateWeatherData(Set<String> features) {
             }
 
             urlStr = urlStr.replace("%QUERY%", StringUtils.trimToEmpty(config.location));
-            logger.debug("URL = {}", urlStr);
+            logger.debug("URL = {}", urlStr.replace("%APIKEY%", "***"));
```

Comment:
```suggestion
            if(logger.isDebugEnabled()){
                logger.debug("URL = {}", urlStr.replace("%APIKEY%", "***"));
            }
```

Reply:
What's the interest to add this test?

Reply:
Since users may not be interested in debug logging, and as such would be filtered out, you should try to minimize wasting computational resources on creating values to be logged.
In this case `String.replace` involves a search and replace, the result of which might not even be used. So in order to prevent unnecessary calls to this operation we just check if that logging level is even enabled in the first place.

Reply:
Ok, done.